HeartbeatLogging/Sources/HeartbeatStorage.swift

```diff
@@ -14,57 +14,84 @@
 
 import Foundation
 
-/// A type that can store and remove hearbeats.
 protocol HeartbeatStorageProtocol {
-  func offer(_ heartbeat: Heartbeat)
-  func flush() -> HeartbeatInfo?
+  typealias HeartbeatInfoTransform = (HeartbeatInfo?) -> HeartbeatInfo?
+
+  func async(_ transform: @escaping HeartbeatInfoTransform)
+  // TODO: Evaluate if async variant of below API is needed.
+  func getAndReset(using transform: HeartbeatInfoTransform?) throws -> HeartbeatInfo?
 }
```

Comment:
Note to readers: In an effort to shift the "whether or not to log a heartbeat" logic to `HeartbeatController`, I needed to make simpler API for this class.